cmd/plakar/subcommands/backup/interactive.go

```diff
@@ -0,0 +1,157 @@
+package backup
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/PlakarKorp/plakar/appcontext"
+	"github.com/PlakarKorp/plakar/events"
+	tea "github.com/charmbracelet/bubbletea"
+)
+
+type tickMsg struct{}
+
+// tick command sends a message after a few ms to update the elapsed time
+func tick() tea.Cmd {
+	return tea.Tick(100*time.Millisecond, func(time.Time) tea.Msg {
+		return tickMsg{}
+	})
+}
+
+type Model struct {
+	ID uint64
+
+	startTime time.Time
+	elapsed   time.Duration
+
+	forceQuit bool
+
+	lastLog string
+
+	countFilesOk     uint64
+	countFilesErrors uint64
+
+	countDirsOk     uint64
+	countDirsErrors uint64
+}
+
+func (m Model) Init() tea.Cmd {
+	return tick()
+}
+
+func (m Model) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
+	switch event := msg.(type) {
+	case tickMsg:
+		m.elapsed = time.Since(m.startTime)
+		return m, tick()
+
+	case events.FileOK:
+		m.countFilesOk++
+		m.lastLog = fmt.Sprintf("%x: %s", event.SnapshotID[:4], event.Pathname)
+
+	case events.FileError, events.PathError:
+		m.countFilesErrors++
+
+	case events.DirectoryOK:
+		m.countDirsOk++
+		m.lastLog = fmt.Sprintf("%x: %s", event.SnapshotID[:4], event.Pathname)
+
+	case events.DirectoryError:
+		m.countDirsErrors++
+
+	case tea.KeyMsg:
+		switch event.String() {
+		case "ctrl+c":
+			m.forceQuit = true
+			return m, tea.Quit
+		}
+
+	case events.Done:
+		m.lastLog = "Done!"
+
+	case tea.QuitMsg:
+		m.lastLog = "Aborted"
+		return m, tea.Quit
+	}
+	return m, nil
+}
+
+func (m Model) View() string {
+	// If nothing was backed up (for example when the directory to backup
+	// doesn't exist), don't show anything.
+	if m.countFilesOk == 0 && m.countFilesErrors == 0 && m.countDirsOk == 0 && m.countDirsErrors == 0 {
+		return ""
+	}
+
+	if m.forceQuit {
+		return fmt.Sprintf("%s Backup aborted\n", crossMark)
+	}
+
+	s := ""
+	s += fmt.Sprintf("Duration: %ds\n", int64(m.elapsed.Seconds()))
+
+	s += fmt.Sprintf("Directories: %s %d", checkMark, m.countDirsOk)
+	if m.countDirsErrors > 0 {
+		s += fmt.Sprintf(" %s %d", crossMark, m.countDirsErrors)
+	}
+	s += "\n"
+
+	s += fmt.Sprintf("      Files: %s %d", checkMark, m.countFilesOk)
+	if m.countFilesErrors > 0 {
+		s += fmt.Sprintf(" %s %d", crossMark, m.countFilesErrors)
+	}
+	s += "\n"
+
+	s += m.lastLog + "\n"
+	return s
```

Comment:
nit: haven't checked how much garbage this produces, but you may consider using a `strings.Builder` here (which you can still feed to `fmt.Fprintf`)

Reply:
done!